app/views/viewlets/inspector-header.js

```diff
@@ -78,13 +78,13 @@ YUI.add('inspector-header-view', function(Y) {
       pojoModel.icon = utils.getIconPath(
           pojoModel.charmUrl, false,
           viewContainerAttrs.charmstore, viewContainerAttrs.env);
+      var name = pojoModel.displayName;
       if (pojoModel.pending) {
         // Check if there is already a service using the default name to
         // trigger the name ux.
         // This regex simply removes the outer parentheses from the
-        // displayName that is set in the ghost-inspector.js updateGhostName
-        // method.  If the regex doesn't match, blow up.  It should match.
-        var name = pojoModel.displayName.match(/^\(([^)]*)\)$/)[1];
+        // displayName that might be set in the model getter.
+        name = name.replace(/^\(/, '').replace(/$\)/, '');
```

Comment:
and then this can just use the same truncate helper vs redoing the logic. Just pass the string in and get it truncated.